### PR TITLE
VACMS-17023 nightly archive workflow

### PR DESCRIPTION
**Pull request by tjheffner**
## Description
Closes #[17023](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/17023).

Adds a separate workflow for archiving a `APP_ENV=prod` build nightly. This re-uses the same bucket that content-build archives into, however we upload into `s3://{{bucket-path}}/next-build/` so there is no fear of overwriting or conflicting with current content-build processes.

This ports over some scripts and custom actions from content-build used for validating the latest passing build.

## Testing done
https://github.com/department-of-veterans-affairs/next-build/actions/runs/8458526038/job/23173013746?pr=478

## Screenshots
the next-build folder in the archive bucket:
Screenshot 2024-03-27 at 2 22 35 PM
https://console.amazonaws-us-gov.com/s3/buckets/vetsgov-website-builds-s3-upload?region=us-gov-west-1&prefix=next-build/&showversions=false

our archived build:
Screenshot 2024-03-27 at 2 22 39 PM

the `9f233085b8b9f7728195a120404cf1704e29201a` name maps to https://github.com/department-of-veterans-affairs/next-build/commit/9f233085b8b9f7728195a120404cf1704e29201a aka the last commit merged to main that successfully passed our `ci.yml` workflow.

## QA steps
```[tasklist]
- [ ] Automated tests have passed
- [ ] Tugboat environment was generated without errors
- [ ] The archive was successfully uploaded to S3
```

### CODE REVIEW

File: .github/workflows/archive.yml [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        timeout_minutes: 30
+        command: cd main && yarn export --DRUPAL_CLIENT_ID ${{ secrets.PROD_DRUPAL_CLIENT_ID }} --DRUPAL_CLIENT_SECRET ${{ secrets.PROD_DRUPAL_CLIENT_SECRET }} --USE_REDIS=false
+
+    - name: Build sitemap
+      run: cd main && yarn build:sitemap
+
+    - name: Compress built site for archive
+      run: tar -c main/out/ -cf ${{ env.BUILD_TYPE }}.tar.bz2 .
+
+    - name: Show build archive size
```
Comment (timcosgrove) [inline .github/workflows/archive.yml:142]: Is there a reason we do this?

**Review by timcosgrove — APPROVED** [review]
Approved, with one tiny non-blocking question

File: .github/workflows/archive.yml [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+        - 80
+      volumes:
+        - /etc/ssl/certs:/etc/ssl/certs
+    steps:
+
+    - name: Checkout next-build
+      uses: actions/checkout@v4
+      with:
+        repository: department-of-veterans-affairs/next-build
+        ref: ${{ needs.validate-build-status.outputs.TAG }}
```
Comment (tjheffner) [inline .github/workflows/archive.yml:99]: from the latest workflow run:
```
Checking out the ref
  /usr/bin/git checkout --progress --force refs/tags/v0.0.111
```

**Review by jschmidt-civicactions — APPROVED** [review]
Looks reasonable to me.

**Pull request merged**